### PR TITLE
Friendlier download dialog + pairing

### PR DESCRIPTION
**Pull request by pelikhan**
image
image

### CODE REVIEW

File: editor/extension.ts [hunk 1]
```diff
@@ -772,4 +773,105 @@ namespace pxt.editor {
...
+        // Data URIs cannot be used for navigation, for scripting, or to populate frame or iframe elements"
+        const downloadAgain = false // !pxt.BrowserUtils.isIE() && !pxt.BrowserUtils.isEdge();
+        const docUrl = pxt.appTarget.appTheme.usbDocs;
+        const columns = canWebusb ? "eleven" : "sixteen";
+
+        const htmlBody = `
+        <div class="ui grid stackable">
+            ${canWebusb ? `<div class="column five wide" style="background-color: #E2E2E2;">
+                <div class="ui header">${lf("One click download?")}</div>
+                <strong style="font-size:small">${lf("Pair your device to get instant download.")}</strong>
```
Comment (riknoll) [inline editor/extension.ts]: Pair your device to download instantly

File: editor/extension.ts [hunk 2]
```diff
@@ -772,4 +773,105 @@ namespace pxt.editor {
...
+                                    <div class="ui">
+                                        <div class="image">
+                                            <img class="ui medium rounded image" src="./static/download/connect.png" style="margin-bottom:1rem;" />
+                                        </div>
+                                        <div class="content">
+                                            <div class="description">
+                                                <span class="ui purple circular label">1</span>
+                                                <strong>${lf("Connect the micro:bit to your computer with a USB cable")}</strong>
+                                                <br />
+                                                <span style="font-size:small">${lf("Use the miniUSB port on the top of the micro:bit Brick")}</span>
```
Comment (riknoll) [inline editor/extension.ts]: microUSB

File: editor/extension.ts [hunk 3]
```diff
@@ -772,4 +773,105 @@ namespace pxt.editor {
...
+
+        return confirmAsync({
+            header: lf("Download to your micro:bit"),
+            htmlBody,
+            hasCloseIcon: true,
+            hideCancel: true,
+            hideAgree: false,
+            agreeLbl: lf("I got it"),
+            className: 'downloaddialog',
+            buttons: [downloadAgain ? {
```
Comment (riknoll) [inline editor/extension.ts]: Nitpick: can this be broken up into an if statement?

File: editor/extension.ts [hunk 4]
```diff
@@ -772,4 +773,105 @@ namespace pxt.editor {
         s.appendChild(f);
         valueNode.appendChild(s);
     }
+
+    function showUploadInstructionsAsync(fn: string, url: string, confirmAsync: (options: any) => Promise<number>) {
+        let resolve: (thenableOrResult?: void | PromiseLike<void>) => void;
+        let reject: (error: any) => void;
+        const deferred = new Promise<void>((res, rej) => {
```
Comment (riknoll) [inline editor/extension.ts]: Is this referenced anywhere?

File: editor/extension.ts [hunk 5]
```diff
@@ -772,4 +773,105 @@ namespace pxt.editor {
...
+            resolve = res;
+            reject = rej;
+        });
+        const boardName = pxt.appTarget.appTheme.boardName || "???";
+        const boardDriveName = pxt.appTarget.appTheme.driveDisplayName || pxt.appTarget.compile.driveName || "???";
```
Comment (riknoll) [inline editor/extension.ts]: Looks like the drive and board names are hard coded below (which is fine I guess)

File: editor/extension.ts (hunk 3 above)
Comment (pelikhan) [reply editor/extension.ts]: maybe one day but not today.

**Comment by pelikhan** [comment]
This dialog will need another round of edits/reskin. Pushing.

**Pull request merged**